### PR TITLE
Upgrade Trumbowyg to 2.13.0

### PR DESCRIPTION
**Pull request by agriffard**
Added trumbowyg 2.13.0 to package.json

Wysiwyg editor has the default options.

I also added a Trumbowyg HtmlField Editor in which you can specify the options you want and load the plugins.

I bundled only a few plugins that are stable. If I load all of them, there is a dependency on Prism for the highlight that we don't want for example.

There seems to be a remaining problem because the 'insertImage' button does not work when you specify the options manually with the Trumbowyg editor, while it works with the Wysiwyg editor.

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Views/HtmlFieldTrumbowygEditorSettings.Edit.cshtml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+@model OrchardCore.ContentFields.ViewModels.TrumbowygSettingsViewModel
+
+<div id="@Html.IdFor(m => m)" class="field-editor field-editor-trumbowyg">
+    <div class="row">
+        <fieldset class="form-group col-sm-6">
+            <label asp-for="Options">@T["Specifiy the trumbowyg editor options"]</label>
+            <textarea asp-for="Options" class="form-control" rows="25"></textarea>
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.ContentFields/Views/HtmlFieldTrumbowygEditorSettings.Edit.cshtml]: No doc, no example? At least a link in the hint to the example.
Ideally could have been some checkboxes for the available plugins.

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Views/HtmlField-Trumbowyg.Edit.cshtml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    var settings = Model.PartFieldDefinition.Settings.ToObject<HtmlFieldSettings>();
+    var trumbowygSettings = Model.PartFieldDefinition.GetSettings<HtmlFieldTrumbowygEditorSettings>();
+}
+
+<script name="trumbowyg" depends-on="admin" asp-src="/OrchardCore.Resources/Scripts/trumbowyg.min.js" debug-src="/OrchardCore.Resources/Scripts/trumbowyg.js" at="Foot"></script>
+<style name="trumbowyg" asp-src="/OrchardCore.Resources/Styles/trumbowyg.min.css" debug-src="/OrchardCore.Resources/Styles/trumbowyg.css"></style>
+<script name="trumbowyg-plugins" depends-on="admin" asp-src="/OrchardCore.Resources/Scripts/trumbowyg-plugins.min.js" debug-src="/OrchardCore.Resources/Scripts/trumbowyg-plugins.js" at="Foot"></script>
+<style name="trumbowyg-plugins" asp-src="/OrchardCore.Resources/Styles/trumbowyg-plugins.min.css" debug-src="/OrchardCore.Resources/Styles/trumbowyg-plugins.css"></style>
+
+<fieldset class="form-group">
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.ContentFields/Views/HtmlField-Trumbowyg.Edit.cshtml]: no more `fieldset`, just divs

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Views/HtmlField-Trumbowyg.Edit.cshtml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+@model OrchardCore.ContentFields.ViewModels.EditHtmlFieldViewModel
+@using OrchardCore.ContentManagement.Metadata.Models
+@using OrchardCore.ContentFields.Settings;
+
+@{
+    var settings = Model.PartFieldDefinition.Settings.ToObject<HtmlFieldSettings>();
+    var trumbowygSettings = Model.PartFieldDefinition.GetSettings<HtmlFieldTrumbowygEditorSettings>();
+}
+
+<script name="trumbowyg" depends-on="admin" asp-src="/OrchardCore.Resources/Scripts/trumbowyg.min.js" debug-src="/OrchardCore.Resources/Scripts/trumbowyg.js" at="Foot"></script>
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.ContentFields/Views/HtmlField-Trumbowyg.Edit.cshtml]: why `name` AND `src`? It's not invalid, just looking for a reason. Or was it this way before?

File: src/OrchardCore.Modules/OrchardCore.Resources/Assets.json [hunk 4]
```diff
@@ -55,16 +55,36 @@
...
+    "inputs": [
+      "node_modules/trumbowyg/dist/plugins/*/ui/sass/*.scss"
+    ],
+    "output": "wwwroot/Styles/trumbowyg-plugins.css"
+  },
   {
     "inputs": [
-      "Assets/trumbowyg/js/trumbowyg.js",
-      "Assets/trumbowyg/plugins/*/*.js"
+      "node_modules/trumbowyg/dist/trumbowyg.js"
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Resources/Assets.json]: Do we need to also copy the Licence file in the wwwroot folder that contains the generated files?

**Comment by sebastienros** [comment]
Screenshot of the options?

**Comment by agriffard** [comment]
trumbowyg

**Comment by sebastienros** [comment]
Not these options, the editor's ones  ;)

**Comment by agriffard** [comment]
media

**Comment by sebastienros** [comment]
Can you use the default trumbowyg settings in the textbox for new fields, and just a link to the documentation in the hint?

**Comment by sebastienros** [comment]
I will wait for my other PR that fixes the package.json files to be merged. Then you'll need to do `gulp build` after merging dev.

I am thinking about having some task on the CI that would do a `gulp rebuild` and check that nothing is impacted. It would let us detect conflicts. /cc @matiasmolleja

**Comment by jonvee** [comment]
Wow, you did a fair bit of work there! I was just getting frustrated not being able to write simple html and having extraneous word formatting in the markup ;)

**Comment by sebastienros** [comment]
> Can you use the default trumbowyg settings in the textbox for new fields, and just a link to the documentation in the hint?

**Comment by agriffard** [comment]
Replaced by #4651

**Pull request closed without merging**